yt_dlp/downloader/hls.py

```diff
@@ -185,15 +96,13 @@ def is_ad_fragment_end(s):
                     if format_index and discontinuity_count != format_index:
                         continue
                     if frag_index > 0:
-                        self.report_error(
-                            'Initialization fragment found after media fragments, unable to download')
-                        return False
+                        raise InitializationFragmentError()
```

Comment:
this should be caught below and trigger `self.report_error`. At the end, the downloader should be calling report_error and not raise ExtractorError.

Also, it doesnt make sense to derive this from `ExtractorError`. It should be a `DownloadError` or a `YoutubeDLError`